Sources/TuistGenerator/Models/Settings.swift

```diff
@@ -55,9 +55,33 @@ public class Settings: Equatable {
     }
 }
 
+extension Settings {
+    /// Finds the default debug `BuildConfiguration` if it exists, otherwise returns the first debug configuration available.
+    ///
+    /// - Returns: The default debug `BuildConfiguration`
+    func defaultDebugBuildConfiguration() -> BuildConfiguration? {
+        let debugConfigurations = configurations.keys
+            .filter { $0.variant == .debug }
+            .sorted()
+        let defaultConfiguration = debugConfigurations.first(where: { $0 == BuildConfiguration.debug })
+        return defaultConfiguration ?? debugConfigurations.first
+    }
+
+    /// Finds the default release `BuildConfiguration` if it exists, otherwise returns the first release configuration available.
```

Comment:
Is this something which might catch people out if they are not careful? If order is important it might make sense to also surface some of this knowledge in the documentation.